### ISSUE TITLE
TypeError: %b requires bytes, or an object that implements __bytes__, not 'str'

### ISSUE THREAD

**Issue by nicolasbock** [seveas/python-hpilo#167]
Maybe I am doing something stupid, but when I compile and install `hpilo` by hand I get this error when querying an iLO:

~~~
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.5/hpilo_cli", line 334, in <module>
    main()
  File "/usr/lib/python-exec/python3.5/hpilo_cli", line 214, in main
    results = [getattr(ilo, method)(**params)]
  File "/usr/lib64/python3.5/site-packages/hpilo.py", line 959, in get_embedded_health
    process=process)
  File "/usr/lib64/python3.5/site-packages/hpilo.py", line 696, in _info_tag
    header, message = self._request(root)
  File "/usr/lib64/python3.5/site-packages/hpilo.py", line 233, in _request
    self._detect_protocol()
  File "/usr/lib64/python3.5/site-packages/hpilo.py", line 273, in _detect_protocol
    header, data = self._communicate(b'<RIBCL VERSION="2.0"></RIBCL>', ILO_HTTP, save=False)
  File "/usr/lib64/python3.5/site-packages/hpilo.py", line 409, in _communicate
    http_header = self.HTTP_HEADER % (msglen, extra_header)
TypeError: %b requires bytes, or an object that implements __bytes__, not 'str'
~~~

**Comment by seveas** [comment]
Fixed in 4.1

**Comment by nicolasbock** [comment]
Thanks!
